src/slam_toolbox_multirobot.cpp

```diff
@@ -0,0 +1,254 @@
+ *
+ * THE WORK (AS DEFINED BELOW) IS PROVIDED UNDER THE TERMS OF THIS CREATIVE
+ * COMMONS PUBLIC LICENSE ("CCPL" OR "LICENSE"). THE WORK IS PROTECTED BY
+ * COPYRIGHT AND/OR OTHER APPLICABLE LAW. ANY USE OF THE WORK OTHER THAN AS
+ * AUTHORIZED UNDER THIS LICENSE OR COPYRIGHT LAW IS PROHIBITED.
+ *
+ * BY EXERCISING ANY RIGHTS TO THE WORK PROVIDED HERE, YOU ACCEPT AND AGREE TO
+ * BE BOUND BY THE TERMS OF THIS LICENSE. THE LICENSOR GRANTS YOU THE RIGHTS
+ * CONTAINED HERE IN CONSIDERATION OF YOUR ACCEPTANCE OF SUCH TERMS AND
+ * CONDITIONS.
+ *
+ */
+
+#include "slam_toolbox/slam_toolbox_multirobot.hpp"
+
+namespace slam_toolbox
+{
+
+/*****************************************************************************/
+MultiRobotSlamToolbox::MultiRobotSlamToolbox(rclcpp::NodeOptions options)
+: SlamToolbox(options), localized_scan_topic_("/localized_scan")
+/*****************************************************************************/
+{
+    current_ns_ = this->get_namespace() + 1;
+
+    localized_scan_pub_ = this->create_publisher<slam_toolbox::msg::LocalizedLaserScan>(
+    localized_scan_topic_, 10);
+    localized_scan_sub_ = this->create_subscription<slam_toolbox::msg::LocalizedLaserScan>(
+        localized_scan_topic_, 10, std::bind(&MultiRobotSlamToolbox::localizedScanCallback, 
+        this, std::placeholders::_1));
+}
+
+/*****************************************************************************/
+void MultiRobotSlamToolbox::laserCallback(
+  sensor_msgs::msg::LaserScan::ConstSharedPtr scan)
+/*****************************************************************************/
+{
+  // store scan header
+  scan_header = scan->header;
+  // no odom info
+  Pose2 pose;
+  if (!pose_helper_->getOdomPose(pose, scan->header.stamp)) {
+    RCLCPP_WARN(get_logger(), "Failed to compute odom pose");
+    return;
+  }
+
+  // ensure the laser can be used
+  LaserRangeFinder * laser = getLaser(scan);
+
+  if (!laser) {
+    RCLCPP_WARN(get_logger(), "Failed to create laser device for"
+      " %s; discarding scan", scan->header.frame_id.c_str());
+    return;
+  }
+
+  LocalizedRangeScan * range_scan = addScan(laser, scan, pose);
+  if (range_scan != nullptr)
+  {
+    Matrix3 covariance;
+    covariance.SetToIdentity();
+    publishLocalizedScan(scan, laser->GetOffsetPose(), 
+        range_scan->GetOdometricPose(), covariance, scan->header.stamp);
+  }
+}
+
+/*****************************************************************************/
+void MultiRobotSlamToolbox::localizedScanCallback(
+  slam_toolbox::msg::LocalizedLaserScan::ConstSharedPtr localized_scan)
+{
+  std::string scan_ns = localized_scan->scan.header.frame_id.substr(0, 
+                          localized_scan->scan.header.frame_id.find('/'));
+  if (scan_ns == current_ns_) return; // Ignore callbacks from ourself
+
+  sensor_msgs::msg::LaserScan::ConstSharedPtr scan = 
+    std::make_shared<sensor_msgs::msg::LaserScan>(localized_scan->scan);
+  Pose2 pose;
```

Comment:
There's a `x,y,heading` constructor for Pose2 that would make this more concise https://github.com/SteveMacenski/slam_toolbox/blob/ros2/lib/karto_sdk/include/karto_sdk/Karto.h#L2063